bson/src/test/unit/org/bson/BinaryVectorTest.java

```diff
@@ -16,17 +16,96 @@
 
 package org.bson;
 
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.read.ListAppender;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.ValueSource;
+import org.slf4j.LoggerFactory;
 
 import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
 class BinaryVectorTest {
 
+    private ListAppender<ILoggingEvent> logWatcher;
+
+    @BeforeEach
+    void setup() {
+        logWatcher = new ListAppender<>();
+        logWatcher.start();
+        ((Logger) LoggerFactory.getLogger("org.bson.BinaryVector")).addAppender(logWatcher);
+    }
+
+    @AfterEach
+    void teardown() {
+        ((Logger) LoggerFactory.getLogger("org.bson.BinaryVector")).detachAndStopAllAppenders();
+    }
+
+    @DisplayName("Treatment of non-zero ignored bits: 1. Encoding")
+    @Test
+    void encodingWithNonZeroIgnoredBits() {
```

Comment:
[nit] For consistency in this file, lets follow "shouldEncode.." naming.
```suggestion
    void shouldEncodeWithNonZeroIgnoredBits() {
```